### PR TITLE
Blog

### LINKED ISSUE #38: Setup ActiveStorage
Currently, it's set to local. However, we should decide if we use Google Cloud or AWS

### LINKED ISSUE #260: Implement Blog Posts
- [x] Accept new postings over Rails
- [x] Collect title, description, content
- [ ] Display within website

### PR DESCRIPTION
**Pull request by lukasbischof**
closes #260 
closes #38

### CODE REVIEW

File: app/controllers/sys_admins/blog_entries_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module SysAdmins
+  class BlogEntriesController < ApplicationController
```
Comment (lukasbischof) [inline app/controllers/sys_admins/blog_entries_controller.rb:4]: Todo: Not authenticated

File: config/environments/development.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@
   end
 
   # Store uploaded files on the local file system (see config/storage.yml for options).
-  config.active_storage.service = :local
+  config.active_storage.service = :google
```
Comment (lukasbischof) [inline config/environments/development.rb:34]: TODO: Revert

File: .semaphore/semaphore.yml [hunk 3]
```diff
@@ -77,7 +77,7 @@ blocks:
       epilogue:
         on_fail:
           commands:
-            - artifact push job logs/test.log
+            - artifact push job log/test.log
```
Comment (lukasbischof) [inline .semaphore/semaphore.yml:80]: Path was wrong

File: Gemfile [hunk 4]
```diff
@@ -12,16 +12,18 @@ gem 'devise_invitable'
 gem 'dotenv-rails'
 gem 'fillable-pdf'
 gem 'flutie'
+gem 'google-cloud-storage', '~> 1.31', require: false
 gem 'hexapdf'
 gem 'holidays'
 gem 'iban-tools'
+gem 'image_processing'
```
Comment (lukasbischof) [inline Gemfile:19]: Required for ActiveStorage

File: Gemfile [hunk 5]
```diff
@@ -46,7 +48,10 @@ end
 
 group :development, :test do
   gem 'brakeman', require: false
-  gem 'bullet'
+  # TODO: Bullet does not yet support Rails 6.1.3.1, but that Rails version is required that ActiveStorage
+  # works properly with the latest Google Cloud Services
+  # => Reenable as soon as support was added
+  # gem 'bullet'
```
Comment (lukasbischof) [inline Gemfile:54]: Problem was:
Either I upgrade Rails and can use the latest active storage version which works correctly with gcs
or
I downgrade to the last working solution and keep Bullet enabled.

Comment (lukasbischof) [reply Gemfile:54]: Obviously went for former option

File: app/views/layouts/organizations/application.html.slim [hunk 6]
```diff
@@ -8,7 +8,7 @@ html
     = csp_meta_tag
     = javascript_pack_tag 'organizations/application', 'data-turbolinks-track': 'reload'
     = stylesheet_pack_tag 'organizations/application', media: 'all', 'data-turbolinks-track': 'reload'
-    = javascript_tag { render 'layouts/organizations/exports.js' }
+    = javascript_tag { render partial: 'layouts/organizations/exports', formats: [:js] }
```
Comment (lukasbischof) [inline app/views/layouts/organizations/application.html.slim:11]: Fixed deprecation warning

File: config/environments/test.rb [hunk 7]
```diff
@@ -4,7 +4,7 @@
   config.cache_classes = false
   config.action_controller.perform_caching = false
   config.action_view.cache_template_loading = true
-  config.action_view.raise_on_missing_translations = true
+  config.i18n.raise_on_missing_translations = true
```
Comment (lukasbischof) [inline config/environments/test.rb:7]: Fixed deprecation warning

**Review by lukasbischof — COMMENTED** [review]
Review comments

**Comment by lukasbischof** [comment]
Actual display on website will come in separate pr

**Review by TheFehr — APPROVED** [review]

**Pull request merged**